### PR TITLE
feat: match folder icon to reflect contents

### PR DESCRIPTION
**Pull request by PThorpe92**
Indicate folder state with icon #92 

### Draft, until agreement on which folder icons to use for empty directories. 

Currently an outlined-open folder would display an empty folder, personally I think this best represents it.

### Another option

Where would you like me to include tests for this?

EDIT: my fault for the CI mess, I was in a hurry and I knew it would be a draft PR anyway.

### CODE REVIEW

File: src/fs/file.rs [hunk 1]
```diff
@@ -427,6 +427,23 @@ impl<'dir> File<'dir> {
...
+    // The naive approach, as one would think that this info may have been cached.
+    // but as mentioned in the size function comment above, different filesystems
+    // make it difficult to get any info about a dir by it's size, so this may be it.
+    #[cfg(unix)]
+    pub fn is_empty_dir(&self) -> bool {
+        if !self.is_directory() {
+            false
+        } else {
+            match Dir::read_dir(self.path.clone()) {
+         // . & .. are skipped, if the returned iterator has .next(), it's not empty
```
Comment (sbatial) [inline src/fs/file.rs:440]:
```suggestion
                   // . & .. are skipped, if the returned iterator has .next(), it's not empty
```

File: src/fs/file.rs [hunk 2]
```diff
@@ -427,6 +427,23 @@ impl<'dir> File<'dir> {
...
+                Ok(has_files) =>  has_files.files(super::DotFilter::Dotfiles, None, false, false).next().is_none(),
+                Err(_) => false,
+                }
+        }
+    }
```
Comment (sbatial) [inline src/fs/file.rs:445]:
```suggestion
    pub fn is_empty_dir(&self) -> bool {
        if self.is_directory() {
            match Dir::read_dir(self.path.clone()) {
                // . & .. are skipped, if the returned iterator has .next(), it's not empty
                Ok(has_files) => has_files.files(super::DotFilter::Dotfiles, None, false, false).next().is_none(),
                Err(_) => false,
            }
        } else {
            false
        }
    }
```

Comment (sbatial) [reply src/fs/file.rs:445]: This should fix the failing CI.
Otherwise: :+1:

**Review by sbatial — CHANGES_REQUESTED** [review]

**Comment by PThorpe92** [comment]
@sbatial 

I had that saved locally flipped as suggested by clippy, I just didn't want to commit it until I write the tests too. Can you point me in the direction of where I should include tests for this?

**Comment by sbatial** [comment]
> I had that saved locally flipped as suggested by clippy, I just didn't want to commit it until I write the tests too.

Oh, my bad. Just force-push over it if u want

> Can you point me in the direction of where I should include tests for this?

So the tests are currently in the `xtests` folder.
We haven't gotten them to work yet.
I just looked into it a bit more and they are written with another tool by @ogham: [ogham/specsheet](https://github.com/ogham/specsheet).

I think the biggest problem right now is that no one has got the Vagrant setup up and running.
The last 'discussion' was in #117.
You could try and see if the docker setup (from https://github.com/ogham/exa/pull/1230) works on your end if you'd like

**Comment by PThorpe92** [comment]
No worries, I just didn't want people to think I was ignoring it, I just didn't want to push (again) and have the CI run, just to go back and add tests and do it again. Either way I just pulled/rebased so it's all good. 

Ah ok.. I was wondering what the state of all those vagrantfile tests was. I understand the unique situation of testing something like an LS replacement, regular unit tests won't work without being run on every imaginable filesystem. I'll take a look at the dockerfile PR and see if I can get it up and running later tonight. Thanks!

**Comment by mentalisttraceur** [comment]
Fwiw I agree using closed solid folder for non-empty and open outlined folder for empty.

**Comment by cafkafk** [comment]
> Fwiw I agree with closed solid folder for non-empty and open outlined folder for empty.

Having solid for contents and outlined for empty seems the most intuitive.

I think this makes sense. 

That said, if we used closed solid and closed outlined, we could have open be folders with content that isn't staged in git yet. But until (if) we actually implemented that behavior, I'd personally prefer `closed solid -> non-empty` and `open outlined -> empty`.

**Comment by PThorpe92** [comment]
Well I wasn't able to get that dockerfile from the exa PR linked above, but I will keep an eye on that branch for sure. It may be time start a discussion or issue on testing, and whether it is worth it to just start including more unit testing in a separate directory?

>  we could have open be folders with content that isn't staged in git yet.

I like that idea, I wonder if there are two different folder icons (open/closed) that have the .git logo, because that would be a nice to display. But at least could be regular subfolders inside repos.

This PR should be all set then, as the current behavior is outlined/open for empty and solid closed for non-empty.

**Review by cafkafk — APPROVED** [review]
> This PR should be all set then, as the current behavior is outlined/open for empty and solid closed for non-empty.

Local testing confirms this working.



LGTM :+1:

**Pull request merged**